src/middleware/updateValidation.js

```diff
@@ -0,0 +1,46 @@
+const constants = require('../lib/constants');
+
+// eslint-disable-next-line consistent-return
+const validateUpdate = (req, res, next) => {
+  const { password, passwordConfirmation, address, age, phoneNumber } =
+    req.body;
+  const errorsArray = [];
+  const properties = Object.keys(req.body);
+  // eslint-disable-next-line consistent-return
+  properties.forEach((property) => {
+    if (req.body[property] === '') {
```

Comment:
Hey @awiednoor. I believe this chunk would cause an error of not allowing the user to change anything, if there's any `req.body` property that is empty. I think you tried it on Postman, and didn't receive any errors, right? For frontend, all `req.body` properties would be there, you can't just not add a `{ phoneNumber: }`, because it _will_ be there, and if a user doesn't fill any of 'em, s/he would receive the error.

Reply:
If you want to check if a field is empty, or has white spaces or "return"s, you can have a look on [this snippet](https://github.com/RCDD-202203-TUR-BEW/backend-capstone-turkey-lets-share/blob/dev/src/middleware/auth.js#L18-L23). Notice how this one doesn't have a `!` at the beginning.

Plus, can't a user change `username`?

Reply:
you are right they can't be _UNDEFINED_ and cannot be checked for that but what i am checking here is to not update to an _empty string_
for the _username_ yes you can update it i added that.